pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1274,3 +1273,36 @@ def func(ser):
 
     expected = DataFrame([[1.0]], index=[1])
     tm.assert_frame_equal(res, expected)
+
+
+@pytest.mark.parametrize(
+    "input_data, expected_output",
+    [
+        (  # timedelta
+            {"dtype": "timedelta64[ns]", "values": ["1 day", "3 days", "NaT"]},
+            {"dtype": "timedelta64[ns]", "values": ["2 days"]},
+        ),
+        (  # datetime
+            {
+                "dtype": "datetime64[ns]",
+                "values": ["2021-01-01T00:00", "NaT", "2021-01-01T02:00"],
+            },
+            {"dtype": "datetime64[ns]", "values": ["2021-01-01T01:00"]},
+        ),
+        (  # timezoned data
+            {
+                "dtype": "datetime64[ns]",
```

Comment:
this is ok but not very explicit, instead use `datetime64[ns, UTC]` or similar (and naive timestamps). for that matter #datetime is incorrect as it will cast as well, instead just use naive values and use `pd.to_datetime` to cast

Reply:
Done.